diff --git a/src/index.flow.js b/src/index.flow.js
index b6aa228bd4cabdc6fef4663a6417c492954a65e1..366e05fcbc67e9b272f5aa8745584594c5de3d67 100644
--- a/src/index.flow.js
+++ b/src/index.flow.js
@@ -143,7 +143,9 @@ exports.runServer = async (
     watch,
   });
   for (const handler of unstable_extraMiddleware ?? []) {
-    serverApp.use(handler);
+    if (handler != null) {
+      serverApp.use(handler);
+    }
   }
   serverApp.use(middleware);
   let httpServer;
